### PR TITLE
Add delimited payload token filter #7978

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
Add delimited payload token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/7978

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@udabhas Could you review this PR for technical accuracy?

**Comment by vagimeli** [comment]
@varun-lodaya The documentation is awaiting tech review and approval, which is delaying progress. Could you please suggest alternative reviewers who can assist with this task in a timely manner? We're eager to move this forward. Thank you.

**Comment by vagimeli** [comment]
> @varun-lodaya The documentation is awaiting tech review and approval, which is delaying progress. Could you please suggest alternative reviewers who can assist with this task in a timely manner? We're eager to move this forward. Thank you.

@varun-lodaya This is over a month old. We need tech review approval to move it forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

**Review by kolchfa-aws — COMMENTED** [review]
Thank you, @AntonEliatra! Please see my changes and let me know if you have any questions.

File: _analyzers/token-filters/delimited-payload.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+      "type": "word",
+      "position": 2
+    }
+  ]
+}
+```
+
+## Example with stored payload
+
+You can configure the payload to be returned in the response by creating an index that stores vectors, using `term_vector` set to `with_positions_payloads` or `with_positions_offsets_payloads` in mappings of the index, see following example:
```
[the remark this answers is not in the document]
Comment (AntonEliatra) [reply _analyzers/token-filters/delimited-payload.md:99]: @kolchfa-aws I'll reword this slightly

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @AntonEliatra! LGTM

File: _analyzers/token-filters/delimited-payload.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
+---
+layout: default
+title: Delimited payload
+parent: Token filters
+nav_order: 90
+---
+
+# Delimited payload token filter
+
+The `delimited_payload` token filter is used to parse tokens containing payloads during the analysis process. For example, the string `red|1.5 fast|2.0 car|1.0` is parsed into tokens `red` (with a payload of `1.5`), `fast` (with a payload of `2.0), and `car` (with a payload of `1.0`). This is particularly useful when your tokens have additional associated data (like weights, scores, or other numeric values) that you can use for scoring or custom query logic. The filter can handle different types of payloads, including integer, float, and strings. and attach payloads (extra metadata) to tokens
```
Comment (natebower) [inline _analyzers/token-filters/delimited-payload.md:10]:
```suggestion
The `delimited_payload` token filter is used to parse tokens containing payloads during the analysis process. For example, the string `red|1.5 fast|2.0 car|1.0` is parsed into the tokens `red` (with a payload of `1.5`), `fast` (with a payload of `2.0`), and `car` (with a payload of `1.0`). This is particularly useful when your tokens include additional associated data (like weights, scores, or other numeric values) that you can use for scoring or custom query logic. The filter can handle different types of payloads, including integers, floats, and strings, and attach payloads (extra metadata) to tokens.
```

File: _analyzers/token-filters/delimited-payload.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
```
Comment (natebower) [inline _analyzers/token-filters/delimited-payload.md:11]: Above: Confirm that my changes to the last sentence retain technical accuracy.

File: _analyzers/token-filters/delimited-payload.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+## Parameters
+
+The `delimited_payload` token filter has two parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`encoding` | Optional | String | Specifies the data type of the payload attached to the tokens. This determines how the payload data is interpreted during analysis and querying.<br>Valid values are:<br><br>- `float`: The payload is interpreted as a 32-bit floating-point number using IEEE 754 format (for example, `2.5` in `car|2.5`).<br>- `identity`: The payload is interpreted as a sequence of characters (for example, in `user|admin`, `admin` is interpreted as a string).<br>- `int`: The payload is interpreted as a 32-bit integer (for example, `1` in `priority|1`).<br> Default is `float`.
+`delimiter` | Optional | String | Specifies the character that separates the token from its payload in the input text. Default is the pipe character (`|`).
+
+## Example without stored payload
```
Comment (natebower) [inline _analyzers/token-filters/delimited-payload.md:25]:
```suggestion
## Example without a stored payload
```

File: _analyzers/token-filters/delimited-payload.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+The `delimited_payload` token filter is used to parse tokens containing payloads during the analysis process. For example, the string `red|1.5 fast|2.0 car|1.0` is parsed into tokens `red` (with a payload of `1.5`), `fast` (with a payload of `2.0), and `car` (with a payload of `1.0`). This is particularly useful when your tokens have additional associated data (like weights, scores, or other numeric values) that you can use for scoring or custom query logic. The filter can handle different types of payloads, including integer, float, and strings. and attach payloads (extra metadata) to tokens
+
+When analyzing text, the `delimited_payload` token filter parses each token, extracts the payload, and attaches it to the token. This payload can later be used in queries to influence scoring, boosting, or other custom behaviors.
+
+Payloads are stored as Base64-encoded strings. By default, payloads are not returned in the query response along with the tokens. To return the payloads, you must configure additional parameters. For more information, see [Example with stored payload]({{site.url}}{{site.baseurl}}/analyzers/token-filters/delimited-payload/#example-with-stored-payload).
```
Comment (natebower) [inline _analyzers/token-filters/delimited-payload.md:14]:
```suggestion
Payloads are stored as Base64-encoded strings. By default, payloads are not returned in the query response along with the tokens. To return the payloads, you must configure additional parameters. For more information, see [Example with a stored payload]({{site.url}}{{site.baseurl}}/analyzers/token-filters/delimited-payload/#example-with-stored-payload).
```

File: _analyzers/token-filters/delimited-payload.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+      "start_offset": 17,
+      "end_offset": 24,
+      "type": "word",
+      "position": 2
+    }
+  ]
+}
+```
+
+## Example with stored payload
```
Comment (natebower) [inline _analyzers/token-filters/delimited-payload.md:97]:
```suggestion
## Example with a stored payload
```

File: _analyzers/token-filters/delimited-payload.md [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+PUT /visible_payloads/_doc/1
+{
+  "text": "red|1.5 fast|2.0 car|1.0"
+}
+```
+{% include copy-curl.html %}
+
+## Generated tokens
+
+Use the following request to examine the tokens generated using the created analyzer:
```
Comment (natebower) [inline _analyzers/token-filters/delimited-payload.md:146]:
```suggestion
Use the following request to examine the tokens generated using the analyzer:
```

File: _analyzers/token-filters/index.md [hunk 8]
```diff
@@ -23,7 +23,7 @@ Token filter | Underlying Lucene token filter|  Description
...
-`delimited_payload` | [DelimitedPayloadTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/payloads/DelimitedPayloadTokenFilter.html) | Separates a token stream into tokens with corresponding payloads, based on a provided delimiter. A token consists of all characters before the delimiter, and a payload consists of all characters after the delimiter. For example, if the delimiter is `|`, then for the string `foo|bar`, `foo` is the token and `bar` is the payload. 
+[`delimited_payload`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/delimited-payload/) | [DelimitedPayloadTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/payloads/DelimitedPayloadTokenFilter.html) | Separates a token stream into tokens with corresponding payloads, based on a provided delimiter. A token consists of all characters before the delimiter, and a payload consists of all characters after the delimiter. For example, if the delimiter is `|`, then for the string `foo|bar`, `foo` is the token and `bar` is the payload.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:26]:
```suggestion
[`delimited_payload`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/delimited-payload/) | [DelimitedPayloadTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/payloads/DelimitedPayloadTokenFilter.html) | Separates a token stream into tokens with corresponding payloads, based on a provided delimiter. A token consists of all characters preceding the delimiter, and a payload consists of all characters following the delimiter. For example, if the delimiter is `|`, then for the string `foo|bar`, `foo` is the token and `bar` is the payload. 
```

**Review by natebower — COMMENTED** [review]
@AntonEliatra @kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _analyzers/token-filters/delimited-payload.md (hunk 3 above)
Comment (AntonEliatra) [reply _analyzers/token-filters/delimited-payload.md:11]: This still makes sense

**Pull request merged**